### ISSUE TITLE
Dataset preparation shell files are not working

### ISSUE THREAD

**Issue by ug-kim** [yangcaoai/CoDA_NeurIPS2023#18]
Hi, thank you for sharing this awesome work.

I want to download your  ov-setting datasets.
However, `bash data_download.sh` is not working now.
Could you check this shell file, especially for the onedrive link?

When I tried to this shell file, I got this error message.
{"error_description":"Invalid JWT token. The token is expired."}

Thank you in advance!

**Comment by yangcaoai** [comment]
Hi, thanks for your interest!

`data_download.sh' has been updated. Please try again :)

**Comment by ug-kim** [comment]
Awesome! Thank you so much for your quick response :)

**Comment by ug-kim** [comment]
I am sorry for bothering you. I just tried your fixed download scripts, but they are not working.
The error message is the same as before.

{"error_description":"Invalid JWT token. The token is expired."}

Could you check one more time? I think the access token expired very shortly.

**Comment by yangcaoai** [comment]
> I am sorry for bothering you. I just tried your fixed download scripts, but they are not working. The error message is the same as before.
> 
> {"error_description":"Invalid JWT token. The token is expired."}
> 
> Could you check one more time? I think the access token expired very shortly.

Yes, the access token expired very shortly(maybe several hours) recently, which may be the new feature of Onedrive. In this case, You may download via the Onedrive links I provide in the Readme.

**Comment by ug-kim** [comment]
Thank you for your kind response.
I now try to download it via OneDrive!

And I have a question about the OV setting dataset configuration.
In the README, you mentioned this.

"To achieve the OV setting, we re-organize the original [ScanNet](https://github.com/facebookresearch/votenet/tree/main/scannet) and [SUN RGB-D](https://github.com/facebookresearch/votenet/tree/main/sunrgbd) and adopt annotations of more categories."

How is it different from the original dataset (ScanNet v2 and SUN RGB-D)? I'm curious about the number of datasets, etc (I think there is some scene sub-sampling?!)

**Comment by yangcaoai** [comment]
> Thank you for your kind response. I now try to download it via OneDrive!
> 
> And I have a question about the OV setting dataset configuration. In the README, you mentioned this.
> 
> "To achieve the OV setting, we re-organize the original [ScanNet](https://github.com/facebookresearch/votenet/tree/main/scannet) and [SUN RGB-D](https://github.com/facebookresearch/votenet/tree/main/sunrgbd) and adopt annotations of more categories."
> 
> How is it different from the original dataset (ScanNet v2 and SUN RGB-D)? I'm curious about the number of datasets, etc (I think there is some scene sub-sampling?!)

The difference is that we utilize annotations of more novel categories. And we have fixed some typo mistakes in the category name. As for the scene number, we didn't conduct sub-sampling. It should be the same as the original datasets.

**Comment by ug-kim** [comment]
Hi, the author! Thank you for your kind responses.

I have some questions about the scanner dataset.
I finished downloading your OV setting dataset, but I have some confusion about this.

1. The total number of ScanNet scenes is 1027 for me, but the paper denoted it is about 1200. Is it correct to download the full dataset?
2. I think the frame is skipped compared to the original ScanNet dataset. For example, the original ScanNet has the sequence number "0, 1, 2, ...", but your dataset has only "0, 100, 200, ...". Is it correct to subsample the sequence of original ScanNet scenes?
3. Following question number 2, with their subsampled sequence, which is "0, 100, 200, ..." I think the point clouds are also divided compared to the original ScanNet. How did you split the original full scene point cloud by the sequences ("0, 100, 200, ...") in the one scene? And why did you split the one scene with the sequences?"

Thank you in advance!

**Comment by yangcaoai** [comment]
> Hi, the author! Thank you for your kind responses.
> 
> I have some questions about the scanner dataset. I finished downloading your OV setting dataset, but I have some confusion about this.
> 
> 1. The total number of ScanNet scenes is 1027 for me, but the paper denoted it is about 1200. Is it correct to download the full dataset?
> 2. I think the frame is skipped compared to the original ScanNet dataset. For example, the original ScanNet has the sequence number "0, 1, 2, ...", but your dataset has only "0, 100, 200, ...". Is it correct to subsample the sequence of original ScanNet scenes?
> 3. Following question number 2, with their subsampled sequence, which is "0, 100, 200, ..." I think the point clouds are also divided compared to the original ScanNet. How did you split the original full scene point cloud by the sequences ("0, 100, 200, ...") in the one scene? And why did you split the one scene with the sequences?"
> 
> Thank you in advance!

Hi, Yu-Ji
(1) If you download the datasets using the link we provided in the readme, it will be correct and fair to compare with our method.
(2) We use the scannet25k, which is a smaller version with fewer frames and provided by the [original Scannet benchmark](https://kaldir.vc.in.tum.de/scannet_benchmark/documentation).
(3) Following [OV-3DET](https://openaccess.thecvf.com/content/CVPR2023/papers/Lu_Open-Vocabulary_Point-Cloud_Object_Detection_Without_3D_Annotation_CVPR_2023_paper.pdf), we compute the point cloud for every single view from depth map on ScanNet, so that the point clouds correspond one-to-one with the single-view images.

**Comment by ug-kim** [comment]
Thank you for your kind response!

Can I get the dataset processing code to get unprotected point clouds and 3D bbox depending on intrinsic? 
In the OV-3DET, [Data-Maker](https://github.com/lyhdet/OV-3DET/tree/main/Data_Maker) code is similar, but I think I cannot use this directly because the novel class definition between OV-3DET and CoDA is different.

It will be helpful to compare your method to my one with my another dataset!

**Comment by yangcaoai** [comment]
> Thank you for your kind response!
> 
> Can I get the dataset processing code to get unprotected point clouds and 3D bbox depending on intrinsic? In the OV-3DET, [Data-Maker](https://github.com/lyhdet/OV-3DET/tree/main/Data_Maker) code is similar, but I think I cannot use this directly because the novel class definition between OV-3DET and CoDA is different.
> 
> It will be helpful to compare your method to my one with my another dataset!

Hi Yu-Ji, I'm sorry that according to the company's current policy, I cannot send codes externally. However, you can modify the script based on [Data-Maker](https://github.com/lyhdet/OV-3DET/tree/main/Data_Maker). The category definition can be referenced at:

https://github.com/yangcaoai/CoDA_NeurIPS2023/blob/main/datasets/scannet_anonymous_aligned_image_with_novel_cate_confi.py#L76 and
https://github.com/yangcaoai/CoDA_NeurIPS2023/blob/main/scripts/coda_scannet_stage2.sh#L30

If you’re still working with ScanNet and SUNRGB-D, I recommend downloading the data I provided for comparison. For datasets other than ScanNet and SUNRGB-D, you may need to reasonably define the base categories and novel categories.

**Comment by Louma20** [comment]
> Hi, thank you for sharing this awesome work.
> 
> I want to download your ov-setting datasets. However, `bash data_download.sh` is not working now. Could you check this shell file, especially for the onedrive link?
> 
> When I tried to this shell file, I got this error message. {"error_description":"Invalid JWT token. The token is expired."}
> 
> Thank you in advance!

Hello @yangcaoai, you are using tempauth, so now I am getting the same issue {"error_description":"Invalid JWT token. The token is expired."}

**Comment by yangcaoai** [comment]
> > Hi, thank you for sharing this awesome work.
> > I want to download your ov-setting datasets. However, `bash data_download.sh` is not working now. Could you check this shell file, especially for the onedrive link?
> > When I tried to this shell file, I got this error message. {"error_description":"Invalid JWT token. The token is expired."}
> > Thank you in advance!
> 
> Hello [@yangcaoai](https://github.com/yangcaoai), you are using tempauth, so now I am getting the same issue {"error_description":"Invalid JWT token. The token is expired."}

Hi, as I mentioned above, the access token expired very shortly(maybe several hours) recently, which may be the new feature of Onedrive. In this case, You may download via the Onedrive links I provide in the Readme instead of data_download.sh
